packages/typespec-client-generator-core/src/types.ts

```diff
@@ -353,9 +353,17 @@ export function getSdkUnionWithDiagnostics(
     return diagnostics.wrap(clientType);
   }
 
-  const unionAsEnum = diagnostics.pipe(getUnionAsEnum(type));
-  if (unionAsEnum) {
-    return diagnostics.wrap(getSdkUnionEnum(context, unionAsEnum, operation));
+  // judge if the union can be converted to enum
```

Comment:
I would say we don't need to judge if it's specifically a union of enum, just that if it's named